### ISSUE TITLE
Contest Analysis Page Hangs a Lot!

### ISSUE THREAD

**Issue by Nayaker** [Nayaker/AlgoListed#113]
The contest analysis page is currently experiencing slowdowns because it calls a rather resource-intensive API and involves a lot of canvas work with Chart.js. 

I'm actively seeking a solution to address this issue.

**_#help-required_**

@jatingodnani please have a look!

**Comment by jatingodnani** [comment]
I think code splitting will work

**Comment by Nayaker** [comment]
Can we get on a google meet for 2 mins?

**Comment by jatingodnani** [comment]
I will today read the whole code.we can schedule meet tomm

**Comment by Nayaker** [comment]
Ok sure, do let me know the timings.

**Comment by jatingodnani** [comment]
Can you pls tell me is that you are calling the api for data to use in chartjs everytime when page load

**Comment by Nayaker** [comment]
YES!

**Comment by jatingodnani** [comment]
Beacuse same api calling and giving same result .we need to use useMemo or callback for optimization

**Comment by Nayaker** [comment]
Ok, but the API parameters matter as well! like the particular contest we are looking for - can we manage that?

**Comment by jatingodnani** [comment]
Yes,suppose if we are calling contest 361 api frequently.if use usememo we can already store the data in this

**Comment by Nayaker** [comment]
Ok sure, I think we can do this!

**Comment by jatingodnani** [comment]
useMemo is not a ssolution,should i do code splitting and break the code into pieces?

**Comment by Nayaker** [comment]
> useMemo is not a ssolution,should i do code splitting and break the code into pieces?

Yeah sure! @jatingodnani 
However, I'd like to make a few adjustments before pushing the code. I'll notify you as soon as I've completed them, which should be within the next hour. 
Additionally, as I mentioned yesterday, this is a complex issue, and I think it would be beneficial to have a discussion on Google Meet. Could you please let me know your availability for scheduling a meeting?

**Comment by jatingodnani** [comment]
To be honest,i dont understand why it is hapenning because in code we only calling two api

**Comment by Nayaker** [comment]
The API is very heavy! If you call it on Postman or something you will realize that. But since the API is made by me on the flask, I can make some required changes as well! That's why wanted a discussion - but if you are busy then it's fine. No issues.

**Comment by jatingodnani** [comment]
I am currently present in collage,will do meet at night.
So this is not  official leetcode api how to get the data from leetcode

**Comment by Nayaker** [comment]
> I am currently present in collage,will do meet at night. So this is not official leetcode api how to get the data from leetcode

Sure, that sounds good. Let's schedule a meeting for tonight.

You're absolutely right; this isn't the official LeetCode API since they don't offer this data. I've customized it myself using Python's Flask and a few other technologies.

**Comment by jatingodnani** [comment]
You are scrapping data from leetcode??if yes,then whivh tool you are using

**Comment by Nayaker** [comment]
Actually, my friend @sailikpandey22 is doing it, I think he is using python beautiful soup and a few other stuff!

**Comment by Nayaker** [comment]
<img width="529" alt="Screenshot 2023-10-06 at 2 03 21 AM" src="https://github.com/Nayaker/AlgoListed/assets/93304796/58bf20f8-876a-433a-96df-aacbff39c5a6">

@jatingodnani I investigated why this issue was happening and found, that the pull request you submitted included some code that was causing an infinite loop. However, after I made the change of replacing [retrivelocalstorage] with an empty [], the issue with the infinite loop was resolved. As a result, we no longer experience any hanging or freezing. :)
